src/plugins/passwd/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+include (LibAddMacros)
+try_compile(HAS_FGETPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasfgetpwent.c")
+if(HAS_FGETPWENT)
+    add_definitions(-DHAS_FGETPWENT)
+endif()
+try_compile(HAS_PUTPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasputpwent.c")
+if(HAS_PUTPWENT)
+    add_definitions(-DHAS_PUTPWENT)
```

Comment:
Please do not add them globally, use `COMPILE_DEFINITIONS` as argument in `add_plugin`.

(In one phase would be enough, I think `try_compile` in this setup will be executed multiple times?)